### DISCUSSION TITLE
[cosign] change over from exec to lib implementation

### DISCUSSION THREAD

**Discussion by dweomer** [hauler-dev/hauler#358] [Ideas] [answered]
Since the advent of https://github.com/ranchergovernment/carbide combined with `hauler store sync --products` functionality we, @amartin120 and I, had chosen to leverage our fork of https://github.com/hauler-dev/cosign via `exec` underneath the hood. We now wish to rescind this and move to leveraging our cosign fork as a library.

See also:
- https://github.com/google/go-containerregistry#partial

**Comment by dweomer** [comment]
![image](https://github.com/user-attachments/assets/07774a95-65ab-476e-9fd5-f7e754c75c55)

**Comment by zackbradys** [comment, marked as answer]
Initial completed in https://github.com/hauler-dev/hauler/pull/383!
